### PR TITLE
Issue #14631: Updated LINK_LITERAL to new AST format

### PR DESCRIPTION
**Pull request by Rohanraj123**
Related #14631 

```
rohanjha@rohanjha-IdeaPad-3-15IIL05-Ua:~/Desktop$ cat Test.java
public class Test {

    /**
     * A simple method that uses {@inheritDoc} inline tag.
     *
     * <p>This method demonstrates the use of {@inheritDoc} tag.</p>
     *
     * {@inheritDoc}
     *
     * @param p1 The first number
     * @return The input number
     */
    public int exampleMethod(int p1) {
        return p1;
    }
    }
rohanjha@rohanjha-IdeaPad-3-15IIL05-Ua:~/Desktop$ java -jar checkstyle-10.17.0-all.jar -J Test.java | sed "s/\[[0-9]\+:[0-9]\+\]//g"
COMPILATION_UNIT -> COMPILATION_UNIT 
`--CLASS_DEF -> CLASS_DEF 
    |--MODIFIERS -> MODIFIERS 
    |   `--LITERAL_PUBLIC -> public 
    |--LITERAL_CLASS -> class 
    |--IDENT -> Test 
    `--OBJBLOCK -> OBJBLOCK 
        |--LCURLY -> { 
        |--METHOD_DEF -> METHOD_DEF 
        |   |--MODIFIERS -> MODIFIERS 
        |   |   |--BLOCK_COMMENT_BEGIN -> /* 
        |   |   |   |--COMMENT_CONTENT -> *\n     * A simple method that uses {@inheritDoc} inline tag.\n     *\n     * <p>This method demonstrates the use of {@inheritDoc} tag.</p>\n     *\n     * {@inheritDoc}\n     *\n     * @param p1 The first number\n     * @return The input number\n      
        |   |   |   |   `--JAVADOC -> JAVADOC 
        |   |   |   |       |--NEWLINE -> \n 
        |   |   |   |       |--LEADING_ASTERISK ->      * 
        |   |   |   |       |--TEXT ->  A simple method that uses  
        |   |   |   |       |--JAVADOC_INLINE_TAG -> JAVADOC_INLINE_TAG 
        |   |   |   |       |   |--JAVADOC_INLINE_TAG_START -> { 
        |   |   |   |       |   |--INHERIT_DOC_LITERAL -> @inheritDoc 
        |   |   |   |       |   `--JAVADOC_INLINE_TAG_END -> } 
        |   |   |   |       |--TEXT ->  inline tag. 
        |   |   |   |       |--NEWLINE -> \n 
        |   |   |   |       |--LEADING_ASTERISK ->      * 
        |   |   |   |       |--NEWLINE -> \n 
        |   |   |   |       |--LEADING_ASTERISK ->      * 
        |   |   |   |       |--TEXT ->   
        |   |   |   |       |--HTML_ELEMENT -> HTML_ELEMENT 
        |   |   |   |       |   `--PARAGRAPH -> PARAGRAPH 
        |   |   |   |       |       |--P_TAG_START -> P_TAG_START 
        |   |   |   |       |       |   |--START -> < 
        |   |   |   |       |       |   |--P_HTML_TAG_NAME -> p 
        |   |   |   |       |       |   `--END -> > 
        |   |   |   |       |       |--TEXT -> This method demonstrates the use of  
        |   |   |   |       |       |--JAVADOC_INLINE_TAG -> JAVADOC_INLINE_TAG 
        |   |   |   |       |       |   |--JAVADOC_INLINE_TAG_START -> { 
        |   |   |   |       |       |   |--INHERIT_DOC_LITERAL -> @inheritDoc 
        |   |   |   |       |       |   `--JAVADOC_INLINE_TAG_END -> } 
        |   |   |   |       |       |--TEXT ->  tag. 
        |   |   |   |       |       `--P_TAG_END -> P_TAG_END 
        |   |   |   |       |           |--START -> < 
        |   |   |   |       |           |--SLASH -> / 
        |   |   |   |       |           |--P_HTML_TAG_NAME -> p 
        |   |   |   |       |           `--END -> > 
        |   |   |   |       |--NEWLINE -> \n 
        |   |   |   |       |--LEADING_ASTERISK ->      * 
        |   |   |   |       |--NEWLINE -> \n 
        |   |   |   |       |--LEADING_ASTERISK ->      * 
        |   |   |   |       |--TEXT ->   
        |   |   |   |       |--JAVADOC_INLINE_TAG -> JAVADOC_INLINE_TAG 
        |   |   |   |       |   |--JAVADOC_INLINE_TAG_START -> { 
        |   |   |   |       |   |--INHERIT_DOC_LITERAL -> @inheritDoc 
        |   |   |   |       |   `--JAVADOC_INLINE_TAG_END -> } 
        |   |   |   |       |--NEWLINE -> \n 
        |   |   |   |       |--LEADING_ASTERISK ->      * 
        |   |   |   |       |--NEWLINE -> \n 
        |   |   |   |       |--LEADING_ASTERISK ->      * 
        |   |   |   |       |--WS ->   
        |   |   |   |       |--JAVADOC_TAG -> JAVADOC_TAG 
        |   |   |   |       |   |--PARAM_LITERAL -> @param 
        |   |   |   |       |   |--WS ->   
        |   |   |   |       |   |--PARAMETER_NAME -> p1 
        |   |   |   |       |   |--WS ->   
        |   |   |   |       |   `--DESCRIPTION -> DESCRIPTION 
        |   |   |   |       |       |--TEXT -> The first number 
        |   |   |   |       |       `--NEWLINE -> \n 
        |   |   |   |       |--LEADING_ASTERISK ->      * 
        |   |   |   |       |--WS ->   
        |   |   |   |       |--JAVADOC_TAG -> JAVADOC_TAG 
        |   |   |   |       |   |--RETURN_LITERAL -> @return 
        |   |   |   |       |   |--WS ->   
        |   |   |   |       |   `--DESCRIPTION -> DESCRIPTION 
        |   |   |   |       |       |--TEXT -> The input number 
        |   |   |   |       |       |--NEWLINE -> \n 
        |   |   |   |       |       `--TEXT ->       
        |   |   |   |       `--EOF -> <EOF> 
        |   |   |   `--BLOCK_COMMENT_END -> */ 
        |   |   `--LITERAL_PUBLIC -> public 
        |   |--TYPE -> TYPE 
        |   |   `--LITERAL_INT -> int 
        |   |--IDENT -> exampleMethod 
        |   |--LPAREN -> ( 
        |   |--PARAMETERS -> PARAMETERS 
        |   |   `--PARAMETER_DEF -> PARAMETER_DEF 
        |   |       |--MODIFIERS -> MODIFIERS 
        |   |       |--TYPE -> TYPE 
        |   |       |   `--LITERAL_INT -> int 
        |   |       `--IDENT -> p1 
        |   |--RPAREN -> ) 
        |   `--SLIST -> { 
        |       |--LITERAL_RETURN -> return 
        |       |   |--EXPR -> EXPR 
        |       |   |   `--IDENT -> p1 
        |       |   `--SEMI -> ; 
        |       `--RCURLY -> } 
        `--RCURLY -> } 
```

### CODE REVIEW

File: src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java [hunk 1]
```diff
@@ -526,7 +526,7 @@ public final class JavadocTokenTypes {
     public static final int LINK_LITERAL = JavadocParser.LINK_LITERAL;
 
     /**
-     * '@inheritDoc' literal in {&#64;inheritDoc} Javadoc inline tag.
+     * '@inheritDoc' literal in {inheritDoc} Javadoc inline tag.
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java:529]: Please revert it, we need `@`
https://checkstyle.org/apidocs/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.html#INHERIT_DOC_LITERAL

File: src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java [hunk 2]
```diff
@@ -535,15 +535,14 @@ public final class JavadocTokenTypes {
...
      * <b>Tree:</b>
-     * <pre>
-     * <code>  |--JAVADOC_INLINE_TAG[1x0] : [{&#64;inheritDoc}]
-     *            |--JAVADOC_INLINE_TAG_START[1x0] : [{]
-     *            |--INHERIT_DOC_LITERAL[1x1] : [@inheritDoc]
-     *            |--JAVADOC_INLINE_TAG_END[1x12] : [}]
-     * </code>
-     * </pre>
+     * <pre>{@code
+     *   |--JAVADOC_INLINE_TAG -> JAVADOC_INLINE_TAG
```
Comment (romani) [inline src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java:541]: Please remove leading `|--` and shift whole block more to left.

**Review by romani — CHANGES_REQUESTED** [review]
Item:

File: src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java (hunk 1 above)
Comment (Rohanraj123) [reply src/main/java/com/puppycrawl/tools/checkstyle/api/JavadocTokenTypes.java]: done !

**Review by romani — APPROVED** [review]
ok to merge

**Pull request merged**